src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+        args['run'] = runNumber
+        args['lumi'] = 0
+        args['fileformat'] = 1
+
+        msg = "HTTP Upload is about to start:\n"
+        msg += " => URL: %s\n" % self.registerURL
+        msg += " => Filename: %s\n" % args['file']
+        logging.info(msg)
+
+        try:
+            logException = True
+            for registerURL in self.registerURL.split(';'):
+                result = self._register(registerURL, [args])
+                msg = '  Status code: %s\n' % result.getcode()
+                if result.getcode() >= 400 or result.getcode() is None:
+                    logException = False
+                    raise Exception(msg)
+                else:
+                    msg = 'HTTP POST to register url finished succesfully with response:\n' + msg
+                    logging.info(msg)
+        except Exception as ex:
+            msg = 'HTTP POST to register url failed! Error:\n%s' % str(ex)
+            if logException:
+                logging.exception(msg)
+            else:
+                logging.error(msg)
+            raise WMExecutionFailure(70319, "DQMUploadFailure", msg)
+
+        return
+
+    def _deleteFromEOS(self, eosFileLFN):
+        """
+        Delete a file uploaded to EOS.
+        I.e.: If we upload a DQM root file but registration failed for some reason
+        """
+        overrides = {}
+        if hasattr(self.step, 'override'):
+            overrides = self.step.override.dictionary_()
+
+        # Note "xrdcp" here will basically map to the XRDCPImpl storage backend
+        # The actual delete command is set in the backend
+        eosStageOutParams = {}
+        eosStageOutParams['command'] = overrides.get('command', "xrdcp")
+        eosStageOutParams['phedex-node'] = overrides.get('eos-phedex-node', "T2_CH_CERN")
+        eosStageOutParams['lfn-prefix'] = overrides.get('eos-lfn-prefix',
+                                                         "root://eoscms.cern.ch/%s" % self.registerEOSPrefix)
+
+        # Switch between old and new stageOut manager.
+        # Use old by default
+        useNewStageOutCode = False
+        if 'newStageOut' in overrides and overrides.get('newStageOut'):
+            useNewStageOutCode = True
+
+        if not useNewStageOutCode:
+            # old style
+            manager = DeleteMgr(**eosStageOutParams)
+            manager.numberOfRetries = self.retryCount
+            manager.retryPauseTime = self.retryDelay
+        else:
+            # new style
+            logging.info("DQMUpload is using new StageOut code to delete files from EOS")
+            manager = NewDeleteMgr(retryPauseTime=self.retryDelay,
+                                   numberOfRetries=self.retryCount,
+                                   **eosStageOutParams)
+
+        logging.info("Deleting LFN: %s", eosFileLFN)
+        eosFileInfo = {'LFN': eosFileLFN,
+                           'PFN': None, # PFNs are assigned in the Delete Manager
```

Comment:
Indentation needs to be fixed.

Reply:
Ah, I wonder how that happened. Thanks, will do